Infiniband patch for OpenMPI < 1.8.6 (WIP)

sylmarien opened the pull request:
Hi, here is the patch relative to this discussion: http://www.open-mpi.org/community/lists/users/2015/05/26913.php

In fact, the novelty of this patch consists only of the last line. So I am not sure about how I should name the file and even if I shouldn't simply put this patch into a new separate file exclusively for this bug (as it will disappear in the future versions of OpenMPI in contrary to the other lines of the patch).

boegel commented:
@sylmarien: this is only useful if the patch is actually used somewhere?

boegel commented:
Jenkins: ok to test

sylmarien commented:
Yes, obviously, I wanted to discuss the patch first.

Basically, we'll need to apply this patch to all OpenMPI prior to 1.8.6.
I'll add it to the said easyconfigs.

boegel commented:
@sylmarien: hold on, this looks like a site customisation for Uni.lu that shouldn't be included centrally?

sylmarien commented:
@boegel Indeed, I put too much in it, I'll modify that too.

Only the `btl_openib_memalign_threshold=12288` part is to be kept in fact.

boegel commented:
@sylmarien: do you have a reference why that is needed? Please include it in the patch as a comment.

sylmarien commented:
I modified the patch, its name and I added it to the easyconfig I know it applies to (all the OpenMPI from 1.6.x to 1.8.5 included). It may apply to 1.4.x too but I am not sure so I let it at that.

As for the comment, I hope it is enough, for a little more complete explanation:
When using the infiniband module, even if there is actually no infiniband on the machine, the value of btl_openib_memalign_threshold was set wrongly (it used another variable which at this point isn't set yet) causing all the memory allocation to be aligned on 32 bits causing important performance losses.
This fix set its default value to the actual default value defined in the documentation.

This bug has been reported and is fixed in the upcoming 1.8.6 version of OpenMPI.

boegel commented:
@sylmarien: I was looking for an URL with 'official' confirmation, but I just noticed it's included in the patch file, i.e. http://www.open-mpi.org/community/lists/users/2015/05/26913.php .

Nice catch by @besserox!

I'll get this PR tested and merged in ASAP.

boegel commented:
synced with develop in https://github.com/ULHPC/easybuild-easyconfigs/pull/2

boegel commented:
merged via #1785
